discovery-provider/src/tasks/index.py

```diff
@@ -446,6 +450,12 @@ def index_blocks(self, db, blocks_list):
         block_number, block_hash, latest_block_timestamp = itemgetter(
             "number", "hash", "timestamp"
         )(block)
+
+        final_poa_block = helpers.get_final_poa_block(shared_config)
+        if final_poa_block and block_number > final_poa_block:
+            logger.info("index.py | skipping block {block_number} past final_poa_block")
+            continue
```

Comment:
shouldn't we have a check here for if we're using ACDC or not? otherwise this check will log and continue even if we're on the new chain?

Reply:
check if we're using ACDC web3? that's really only necessary when we're offsetting by final_poa_block. index.py should always use POA web3 and index_nethermind will always use ACDC web3. 

this is mainly to make sure we only index up to final_poa_block. these blocks will be in ascending order but i'll change continue -> break in case.